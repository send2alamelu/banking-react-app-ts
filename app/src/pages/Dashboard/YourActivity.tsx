import { useEffect, useState } from 'react';
import Divider from '@mui/material/Divider';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';

import useGet from '../../hooks/useGet';
import { getTransactionApiPath } from '../../utils/Api';
import { GetResponse } from '../../types/ApiResponse';
import { TransactionItem } from '../../types/TransactionItem';
import { getDescriptionFromTransactionType } from '../../utils/Transaction';

export default function YourActivity() {
  const getTransaction: GetResponse = useGet(getTransactionApiPath());
  const [transactionList, setTransactionList] = useState([]);

  useEffect(() => {
    const transactions = getTransaction?.response?.data || []
    if (transactions.length > 0) {
      setTransactionList(transactions);
    }

  }, [getTransaction?.response]);

  return (
    <>
      <h1>Your Activity</h1>
      <Divider />
      <TableContainer component={Paper}>
        <Table aria-label="activity table">
          <TableBody>
            {transactionList.map((transaction: TransactionItem) => (
              <TableRow
                key={transaction.amount}
                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
              >
                <TableCell component="th" scope="row">
                  {transaction.date}
                </TableCell>
                <TableCell align="left">{getDescriptionFromTransactionType(transaction)}</TableCell>
                <TableCell align="left">{transaction.amount}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  );
}
