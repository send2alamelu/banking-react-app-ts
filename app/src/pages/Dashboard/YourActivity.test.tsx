import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import YourActivity from './YourActivity';

describe('YourActivity Component', () => {
  afterEach(() => {
    cleanup();
  });

  test('renders YourActivity with children', () => {
    // Arrange.
    const { asFragment } = render(<YourActivity />);
    
    // Assert.
    // expect(getByText(/Your Activity/i)).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});


