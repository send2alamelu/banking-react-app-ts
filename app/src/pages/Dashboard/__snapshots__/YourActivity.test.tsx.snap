// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourActivity Component renders YourActivity with children 1`] = `
<DocumentFragment>
  <h1>
    Your Activity
  </h1>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="activity table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      />
    </table>
  </div>
</DocumentFragment>
`;
